### DISCUSSION TITLE
Lights flashes randomly

### DISCUSSION THREAD

**Discussion by versionDefect** [awawa-dev/HyperHDR#901] [Q&A]
Hi everyone, I've got a Pi 3B (NOT-undervolted) and a D1 Mini ESP32 CP2104 and SK6812  I've soldered the pins D6 to the LEDs, Soldered ground on the controller to the ground on the PSU, and for some reason the lights flicker, and I don't really know what it could be. 

I'm honestly stuck, I don't really know what this issue can be. The one thing I think it could be is the fact that I'm using a longer micro USB cable (6 ft) but its not really possible for me to test it without it. 

https://github.com/user-attachments/assets/0a2e2977-f040-49b0-a19c-b74034c97d2d
https://github.com/user-attachments/assets/9d8276ca-4a57-47cf-a929-011e0a98a1f1

Considering its happening in effects I think its safe to say its something between either HyperHDR or the Controller.

**Comment by awawa-dev** [comment]
Hi
> D1 Mini ESP32 CP2104 and SK6812 I've soldered the pins D6 to the LEDs

Description points to inferior esp8266 not ESP32, HyperSerial8266 doesnt work on that pin so probably WLED? If that's the case it's probably famous bitbang, cannt help with the esp8266 software, they consider esp8266 as depreciated. When it comes to hardware, a level-shifter is missing.

> Soldered ground on the controller to the ground on the PSU

Did you mean "minus" being a pair to +5v on the power supply?

**Reply by versionDefect** [reply]
Yes minus is also soldered to the GND pin on the controller itself. 

Also I don't have HyperSerial8266, i've got HyperSerialESP32, I must've mixed up pins from back when I did use my 8266. I'm using GPIO 2 on the controller board. I've flashed the firmware for an ESP32 so i'm pretty confident thats what it is.

**Reply by awawa-dev** [reply]
D1 Mini type board and D6 pin are typical attributes of esp8266 boards. Anyway you should think about a 3.3V to 5V level shifter if you are sure you connected everything correctly.

**Reply by versionDefect** [reply]
> D1 Mini type board and D6 pin are typical attributes of esp8266 boards. Anyway you should think about a 3.3V to 5V level shifter if you are sure you connected everything correctly. 

Sounds good! Is there a one you'd recommend?

**Reply by awawa-dev** [reply]
If you like DIY solution then SN74AHCT125N : https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter
as an alternative Piromoni and Adafruit boards with already built-in level shifter: https://github.com/awawa-dev/HyperSerialPico?tab=readme-ov-file#recommended-boards-with-a-built-in-level-shifter
